### PR TITLE
Fix build by making LockfileIndexResolver interface less generic

### PR DESCRIPTION
**Pull request by mrnugget**
With only an ID property every resolver implementation matched it, that
broke QA and E2E tests.

## Test plan

- Manual testing

### CODE REVIEW

File: internal/codeintel/dependencies/transport/graphql/lockfile_index_resolver.go [hunk 1]
```diff
@@ -8,13 +8,17 @@ import (
 )
 
 type LockfileIndexResolver struct {
-	executor shared.LockfileIndex
+	lockfile shared.LockfileIndex
 }
 
 func NewExecutorResolver(executor shared.LockfileIndex) *LockfileIndexResolver {
-	return &LockfileIndexResolver{executor: executor}
+	return &LockfileIndexResolver{lockfile: executor}
```
Comment (eseliger) [inline internal/codeintel/dependencies/transport/graphql/lockfile_index_resolver.go:15]: more executor here

**Review by eseliger — APPROVED** [review]

**Review by burmudar — APPROVED** [review]

**Comment by mrnugget** [comment]
Just for context, @burmudar: build failed because of this
screenshot_2022-07-15_14 09 13@2x


Our root resolver was casting a node to a specific type and since the type I introduced `LockfileIndexResolver` was so generic (implementable by adding an `ID() graphql.ID` method) every other resolver implemented it and that tripped everything up.

**Comment by burmudar** [comment]
Thanks for the context 😄

**Comment by mrnugget** [comment]
Merged https://github.com/sourcegraph/sourcegraph/pull/38877 into this one because we need both fixes for the build to run and to merge.

**Pull request merged**